### DISCUSSION TITLE
Windows OS stdpopsim run issues

### DISCUSSION THREAD

**Discussion by smccallum76** [popsim-consortium/stdpopsim#1519] [Q&A] [answered]
No doubt that this issue is on my end, but I have encountered problems with the Windows install of stdpopsim.  No errors are reported upon the install of stdpopsim. However, when importing into a Python script or testing at the command line, numerous errors are reported. Any advice as to how to repair this issue would be greatly appreciated. The workflow that has been followed is copied below:

1. Install SLiM and confirm that SLiM functions at the command line (it does)
2. Create a conda environment
3. Activate the environment
4. install stdpopsim [conda install -c conda-forge stdpopsim]. No errors are reported after the install.
5. Test the install using <stdpopsim --version>. At this point the errors shown below are reported.
![image](https://github.com/popsim-consortium/stdpopsim/assets/44415082/2f832063-70b5-4a7e-9c9d-7c87de0dcbd3)

**Note**: I have attempted to work around this by using Linux/Ubuntu, which partially solves the problem. However, a Windows install would be my preferred platform for an upcoming project.

**Comment by jeromekelleher** [comment, marked as answer]
This is actually an upstream problem in tskit @smccallum76 discussed here: https://github.com/tskit-dev/tskit/issues/2840

Basically, the jsonschema package has made some breaking changes, which we need to catch up with. We'll have a release of tskit out to address shortly.

Installing jsonschema==4.18 before installing stdpopsim should resolve the problem in the short term.

**Reply by smccallum76** [reply]
@jeromekelleher Thanks much for the feedback. I'll follow the workflow you suggested above. (Again, much appreciated, I was hung up on this issue for a while).

**Reply by petrelharp** [reply]
Just to record here: this was fixed in tskit 0.5.6, released Oct 10, 2023: https://github.com/tskit-dev/tskit/issues/2840#issuecomment-1755784251
